File: problems/53/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        best_sum, cur_sum = nums[0], -1
        
        for v in nums:
            cur_sum = v if cur_sum < 0 else cur_sum + v
            best_sum = max(best_sum, cur_sum)
        
        return best_sum

File: problems/53/solution.go
func maxSubArray(nums []int) int {
    bestSum, curSum := nums[0], nums[0]
    
    for _, v := range nums[1:] {
        curSum = max(v, curSum + v)
        bestSum = max(bestSum, curSum)
    }
    
    return bestSum
}

func max(a, b int) int {
    if a > b {
        return a
    }
    
    return b
}